pingpong/server.py

```
@@ -11847,6 +11848,239 @@ async def get_external_logins(request: StateRequest):
+    response_model=schemas.ConnectorConnectResponse,
+)
+async def connect_connector(
+    request: StateRequest,
+    service: str,
+    body: schemas.ConnectorConnectRequest,
+):
+    """Build the provider's authorize URL and return it for the frontend redirect."""
+    try:
+        connector = connectors_pkg.get(service)
+    except connectors_pkg.ConnectorNotRegistered:
+        raise HTTPException(status_code=404, detail="Unknown connector")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)